src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -572,9 +572,26 @@ public static bool IsExtensionMethodAndHasNoInstance(this IInvocationOperation i
             return invocationOperation.TargetMethod.IsExtensionMethod && (invocationOperation.Language != LanguageNames.VisualBasic || invocationOperation.Instance == null);
         }
 
-        public static SyntaxNode GetInstance(this IInvocationOperation invocationOperation)
+        public static IOperation GetInstance(this IInvocationOperation invocationOperation)
+            => invocationOperation.IsExtensionMethodAndHasNoInstance() ? invocationOperation.Arguments[0].Value : invocationOperation.Instance;
+
+        public static SyntaxNode GetInstanceSyntax(this IInvocationOperation invocationOperation)
+            => invocationOperation.GetInstance().Syntax;
+
+        public static ITypeSymbol GetInstanceType(this IOperation operation)
```

Comment:
Instance can be null for static member references.
```suggestion
        public static ITypeSymbol? GetInstanceType(this IOperation operation)
```

Reply:
I couldn't get a repro where this would return null, even with static members:
```cs
using System;
using System.Linq;
using System.Collections.Concurrent;

public class Test
{
    private static ConcurrentDictionary<string, string> _dictionary;

    public bool Foo => _dictionary.Count > 0;
}
```

Reply:
Probably not for your case, but given this is an extension method, some other analyzer might call it from the context of a static method invocation in future.